brian2/core/variables.py

```diff
@@ -1580,7 +1580,7 @@ def __hash__(self):
     @property
     def shape(self):
         if self.ndim == 1:
-            if hasattr(self, "group.start") and hasattr(self, "group.stop"):
+            if hasattr(self, "group") and hasattr(self.group, "start") and hasattr(self.group, "stop"):
```

Comment:
Hah, saved by the testsuite, I did not spot this just by reading :) Usually I write this as `hasattr(getattr(self, "group", None), "start")`, but this would be cumbersome here, since `group` is used twice – I think your approach is the best/clearest.